brave/brave-ios
# Fix #7699: NFT filters and display settings

## Summary of Changes
- Integrates Filters and Display Settings modal into the NFT tab
- NFT tab has a `Hide Unowned` and `Show Network Logo` display setting instead of the `Sort Assets` & `Hide Small Balances` seen on Portfolio.
- Selected accounts & networks are shared with Portfolio

This pull request fixes #7699

## Submitter Checklist:


## Test Plan:

### Test `Show Network Logo`:
1. Have at least 1 NFT visible in NFT tab.
2. Tap filter button on NFT tab to open Filters and Display Settings.
3. Toggle `Show Network Logo` on/off, then tap `Save Changes`.
4. Verify logo is shown on NFTs in grid when enabled and not shown when hidden.

https://github.com/brave/brave-ios/assets/5314553/ee3b8910-22dc-485f-bac5-33da2b3ad1a7


### Test `Hide Unowned`:
1. Have at least 1 NFT that you own, and one NFT that you do not own added to NFT tab.
2. Tap filter button on NFT tab to open Filters and Display Settings.
3. Toggle `Hide Unowned` on/off, then tap `Save Changes`.
4. When enabled, verify only the owned NFT is displayed. When disabled, verify all NFTs are shown (owned and unowned).

https://github.com/brave/brave-ios/assets/5314553/a427725c-26b2-476a-ad19-04d0e5beb95f


### Test `Accounts`:
1. Have at least 1 NFT that you own added to NFT tab.
2. Enable `Hide Unowned` toggle (required for Accounts filters to be used until Grouping support).
3. Tap filter button on NFT tab to open Filters and Display Settings.
4. Tap accounts and de-select the account that owns the NFT, tap back then tap `Save Changes`.
5. Verify the NFT is no longer displayed in the grid.

https://github.com/brave/brave-ios/assets/5314553/619d77f4-3b9f-4c53-bc34-7872d5013f05


### Test `Networks`:
1. Have at least 1 NFT that you own added to NFT tab.
2. Tap filter button on NFT tab to open Filters and Display Settings.
3. De-select the network the NFT belongs to, tap back then tap `Save Changes`.
4. Verify the NFT belonging to the de-selected network is hidden.
5. Repeat from step 2 but de-select all networks, then select the network the NFT belongs to.
6. Verify the NFT belonging to the selected network is shown.

https://github.com/brave/brave-ios/assets/5314553/9cd9da57-42e7-4399-af83-b189e90c724e



## Screenshots:

![NFT Filters   Display Settings - light](https://github.com/brave/brave-ios/assets/5314553/c4381ba6-fd7e-4f01-b0d3-895390e02205) | ![NFT Filters   Display Settings - dark](https://github.com/brave/brave-ios/assets/5314553/d17cf4e3-b3b0-456b-934d-a8061a661ffa)
--|--

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
